File: php/features.php
<?php
declare(strict_types=1);

// ExchangeRates SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ExchangeRatesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ExchangeRatesBaseFeature();
            case "test":
                return new ExchangeRatesTestFeature();
            default:
                return new ExchangeRatesBaseFeature();
        }
    }
}

File: py/features.py
# ExchangeRates SDK feature factory

from feature.base_feature import ExchangeRatesBaseFeature
from feature.test_feature import ExchangeRatesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ExchangeRatesBaseFeature(),
        "test": lambda: ExchangeRatesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
